BaSP-a2022/marta-b-trackgenix-app
# BaSP-TG-44: Assign Employee feature

Ticket: https://trello.com/c/XYPrtSeT/44-project-description-add-edit-delete-assigned-employees-projects

We added a new feature to assign an employee to a Project using React hook form and Modal component. The Modal has 3 inputs: One where you choose from a list of employees by last name. The second input is where you choose which role do you want the employee to have on the project. There are 3 options: Developer, Tech Lead or Quality Assurance. The third and last option is where you write the rate of that employee.

**How to test**:
- Log In as Admin (username: nuevoadmin@gmail.com, password: nuevoadmin).
- Go to 'Projects' section.
- Click on 'Assign' button to display the modal with the form.
- Choose an employee.
- Choose a role.
- Add a rate.
- Click on Assign to assign that employee to a project.
- On the project list, click on 'Employees' to see if the new employee has been added.

Validations: 
- Employee:
"Employee" cannot be empty.
- Role:
"role" must be one of [DEV, QA, TL]
- Rate
"rate" must be a number

**Pending**:
- Add Project Manager role and its features.
- Fix table not fitting the modal.
- Fix validations on backend.
- Edit/Delete assigned employees.

Contributors: @mara666
